### PR TITLE
Feat/dev 1801 gitlab ci

### PR DESCRIPTION
**Pull request by kulgan**
Switch test and build to using gitlab ci. Run tests for multiple python versions since indexd is still used by multiple python versions for testing

### New Features
- Gitlab CI configuratione
- Remove jenkins pipeline and associated files

### Improvements
- Allows database parameters to be configured via environmental variable

### Dependency updates
- run pip-compile using python3.6 to maintain compatibility with other projects using indexd as a dependency

### CODE REVIEW

File: indexd/index/version_data.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
-VERSION = ""
-COMMIT = ""
+COMMIT = "c61ead7c4ca94104b8a8988edf4e5598a84757d8"
+VERSION = "2.11.0-51-gc61ead7"
```
Comment (ProfOak) [inline indexd/index/version_data.py:2]: What are these versions? Do we need to do something special to set them next time?

**Review by ProfOak — APPROVED** [review]

File: indexd/index/version_data.py (hunk 1 above)
Comment (kulgan) [reply indexd/index/version_data.py:2]: This is updated each time we run the docker.sh file. Tried to keep it as is for now, but we may have to come back to it

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    PG_INDEXD_ROOT_USER: test
+    PG_INDEXD_ROOT_PASS: test
+    POSTGRES_DB: indexd_test
+    POSTGRES_USER: test
+    POSTGRES_PASSWORD: test
+    POSTGRES_HOST_AUTH_METHOD: trust
+  before_script:
+    - pip install 'tox < 4'
+    - mkdir -p /usr/share/man/man1
+    - apt update && apt -y install openjdk-8-jdk-headless
```
Comment (qiaouchicago) [inline .gitlab-ci.yml:50]: why do we need jdk?

**Review by qiaouchicago — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (kulgan) [reply .gitlab-ci.yml:50]: To generate the openapi client used in testing

**Pull request merged**